daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -525,8 +536,19 @@ private[lf] case class PartialTransaction(
     context.info match {
       case info: TryContextInfo =>
         // TODO https://github.com/digital-asset/daml/issues/8020
-        //  for now, we just drop the whole rollback part of the tree.
-        copy(context = info.parent.copy(nextChildIdx = context.nextChildIdx))
+        //  the version of a rollback node should be determined from its children.
+        //  in the case of there being no children we can simple drop the entire rollback node.
+        val version: TxVersion = VDev
+        val rollbackNode = Node.NodeRollback(
+          context.children.toImmArray,
+          version,
+        )
+        val nodeId = info.nodeId
+        copy(
+          context = info.parent.addChild(nodeId),
+          nodes = nodes.updated(nodeId, rollbackNode),
+          //nodeSeeds = nodeSeeds :+ (nodeId -> info.nodeSeed), //TODO 8020, need this?
```

Comment:
```suggestion
          nodeSeeds = nodeSeeds.update(nodeId, info.nodeSeed),
```
Lets be simple and add everything naively.